test/e2e/daemon_set.go

```diff
@@ -61,7 +61,15 @@ func clearNodeLabels(c *client.Client) error {
 	for _, node := range nodeList.Items {
 		if len(node.Labels) != 0 {
 			node.Labels = map[string]string{}
-			newNode, err := nodeClient.Update(&node)
+			var newNode *api.Node
+			err = wait.Poll(5*time.Second, 30*time.Second, func() (bool, error) {
+				newNode, err = nodeClient.Update(&node)
+				if err != nil {
```

Comment:
I think that you should be more specific about the error codes on which you retry?